meilisearch-http/src/routes/search.rs

```diff
@@ -28,6 +30,7 @@ struct SearchQuery {
     attributes_to_highlight: Option<String>,
     filters: Option<String>,
     matches: Option<bool>,
+    facet_filters: Option<String>,
```

Comment:
As I understand this line a user cannot send a valid JSON array? He must send a string containing the JSON array itself? I think we must look into serde and probably create our own enum type to be able to accept real JSON arrays.

Reply:
I'm not sure to understand what you mean, but the user sends an array in an url-encoded format, that upon decoding happen to be JSON-like.  If we use serde to parse it, we'll have a "syntactically" correct array, but when I parse it I actually do a lot more than what serde can do, I'd have to go through it once more to make parse keys, and make sure the array is correct. I'm not sure to see how this would be an improvement over what we are doing right now